src/test/regress/expected/intermediate_result_pruning.out

```diff
@@ -762,7 +762,8 @@ ROLLBACK;
 -- We use offset 1 to make sure the result needs to be pulled to the coordinator, offset 0 would be optimized away
 INSERT INTO table_1
 	SELECT * FROM table_2 OFFSET 1;
-DEBUG:  OFFSET clauses are not allowed in distributed INSERT ... SELECT queries
+DEBUG:  cannot push down this subquery
+DETAIL:  Offset clause is currently unsupported when a subquery references a column from another query
```

Comment:
DeferErrorIfUnsupportedSubqueryPushdown assumes that the input query is a subquery, hence the debug message is misleading

Reply:
Yeah, right. But I'm fine with that as of this PR given it is only debug message.

We can definitely consider a follow=up patch to improve the messages, but for now it is good to keep this patch smaller.

@onurctirtir is that OK for you as well?